lib/pf/cmd/pf/service.pm

```diff
@@ -88,15 +88,24 @@ our %ACTION_MAP = (
 
 our $ignore_checkup = $FALSE;
 
+sub _byIndexOrder { 
+    $a->new->orderIndex <=> $b->new->orderIndex;
+}
+
 sub parseArgs {
```

Comment:
To ensure that the sorting is "stable" add the following pragma.
use sort 'stable';